friends/src/App.js

```diff
@@ -0,0 +1,140 @@
+import React, { Component } from 'react';
+import { Route, withRouter } from 'react-router-dom';
+import axios from 'axios';
+
+import FriendList from './components/FriendList';
+import NewFriend from './components/NewFriend';
+import EditFriend from './components/EditFriend';
+
+class App extends Component {
+  constructor() {
```

Comment:
Check out this [article](https://medium.com/technical-credit/things-to-stop-start-doing-in-react-9887b9f6cea6) on why we can now do this in React. Just something to try out in your code that can make it more concise. 
```suggestion
state = {
  friends: [],
  currentFriend: []
};
```